qa/tasks/ceph.py

```diff
@@ -426,12 +426,20 @@ def cephfs_setup(ctx, config):
     # If there are any MDSs, then create a filesystem for them to use
     # Do this last because requires mon cluster to be up and running
     if mdss.remotes:
-        log.info('Setting up CephFS filesystem...')
+        log.info('Setting up CephFS filesystem(s)...')
+        cephfs_configs = config.get('cephfs', {'cephfs': {}})
+        set_allow_multifs = True if len(cephfs_configs) > 1 else False
+
+        for name, cephfs_config in cephfs_configs.items():
```

Comment:
I think this structure will end up being inflexible. Let's change it to:

````
ceph:
  cephfs:
    fs:
    - name: cephfs
      ec_profile: bar
    - name: cephfs2
      ec_profile: bar2
````

And the "default" if `fs` is unspecified is the normal default file system, of course.

Reply:
> I think this structure will end up being inflexible. Let's change it to:
> 
> ```
> ceph:
>   cephfs:
>     fs:
>     - name: cephfs
>       ec_profile: bar
>     - name: cephfs2
>       ec_profile: bar2
> ```
> 
> And the "default" if `fs` is unspecified is the normal default file system, of course.

I don't get why my change is inflexible? Because you always need to enter the file system name? I thought it will help make the config setting unambiguous.

Questions based on your suggestion in the following scenarios where the final config including the overrides looks like this,
Scenario 1:
```
ceph:
  cephfs:
    session_timeout: 30
    fs:
    - name: cephfs
      ec_profile: bar
    - name: cephfs2
      ec_profile: bar2
```
`cephfs` is the default file system.  Do you expect the `session_timeout` config setting to be applied to
 - all filesystems ('cephfs' and 'cephfs2'), or 
 - only default filesystem 'cephfs', or
 - none. The `session_timeout` config is ignored

Scenario 2:
```
ceph:
  cephfs:
   ec_profile: defaultbar
    fs:
    - name: cephfs
      ec_profile: bar
    - name: cephfs2
      ec_profile: bar2
```
`cephfs` is the default file system.  What do you expect the `ec_profile` setting of filesystem `cephfs` and `cephfs2` to be and why?

Reply:
> I don't get why my change is inflexible? Because you always need to enter the file system name? I thought it will help make the config setting unambiguous.

Sorry I wasn't clear. The reason I think it will be inflexible is we won't be able to easily add configurations for global cephfs-related QA changes except in a different namespace from `ceph/cephfs`. Sticking all of the file systems in `ceph/cephfs/fs` is one way to avoid that.

As for how the settings are applied, I'd suggest a hierarchy. If `ceph/cephfs/fs/foo` has `ceph/cephfs/fs/foo/session_timeout`, use that. Otherwise, use `ceph/cephfs/session_timeout`. Otherwise, use the default.